src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -1015,13 +1009,4 @@ inline bool ShenandoahRegionChunkIterator::next(struct ShenandoahRegionChunk *as
   return true;
 }
 
-template<class T>
-inline void ShenandoahVerifyNoYoungRefsClosure::work(T* p) {
-  T o = RawAccess<>::oop_load(p);
-  if (!CompressedOops::is_null(o)) {
-    oop obj = CompressedOops::decode_not_null(o);
-    assert(!_heap->is_in_young(obj), "Found a young ref");
-  }
-}
-
```

Comment:
Very curious. I could have sworn I had used this in some assertions/verification code in card scanning or card verification, but may be that code went away. Does seem to be unused now, so best to get rid of it.